### PR TITLE
Add wavy stereo bond support to cdxml

### LINKED ISSUE #5752: CDXML reader incorrectly sets stereo on crossed double bonds
RDKit: 2022.09.1

I've been playing around with pycdxml from @kienerj, and testing it by roundtripping MOL files against the RDKit's CDXML reader. Here's a failure case which seems to be due to the RDKit reader.

If you open the following MOL file in ChemDraw:
[CHEMBL4303146.mol.txt](https://github.com/rdkit/rdkit/files/9995609/CHEMBL4303146.mol.txt)
image

...and save as CDXML:
[CHEMBL4303146.cdxml.txt](https://github.com/rdkit/rdkit/files/9995611/CHEMBL4303146.cdxml.txt)
...then RDKit's reader sets the stereo on the crossed double bond instead regarding it as unknown. That is,
```
from rdkit import Chem

fnames = ["CHEMBL4303146.fromcd.cdxml"]
for fname in fnames:
    mols = Chem.MolsFromCDXMLFile(fname)
    cansmi = Chem.MolToSmiles(mols[0])
    print(cansmi)
```
gives
```
Cc1cccc(C)c1NC(=O)/N=C1\CCCN1C
```

### PR DESCRIPTION
**Pull request by bp-kelley**
Fixes #5752

### CODE REVIEW

File: Code/GraphMol/FileParsers/CDXMLParser.cpp [hunk 1]
```diff
@@ -422,6 +422,16 @@ bool parse_fragment(RWMol &mol, ptree &frag,
       } else if (bond.display == "WedgedHashBegin" ||
                  bond.display == "WedgedHashEnd") {
         bnd->setBondDir(Bond::BondDir::BEGINWEDGE);
+      } else if (bond.display == "Wavy") {
+	switch(bond.getBondType()) {
```
Comment (bp-kelley) [inline Code/GraphMol/FileParsers/CDXMLParser.cpp:426]: maybe should be an if statement?  Should a wavy bond always be a double bond?  There are some cases where it isn't in the test set (these are obviously incorrect)

Comment (greglandrum) [reply Code/GraphMol/FileParsers/CDXMLParser.cpp:426]: I wouldn't think that it should always be double.
I'm not sure how chemdraw works with this stuff, but in v3k mol files bond CFG=2 is rendered as a crossed bond when applied to a double bond or as a wavy bond if it's single. You definitely need to be able to represent wavy single bonds, but maybe chemdraw does that differently?

Comment (bp-kelley) [reply Code/GraphMol/FileParsers/CDXMLParser.cpp]: I think this has the right logic now for wavy bonds.

File: Code/GraphMol/FileParsers/CDXMLParser.cpp [hunk 2]
```diff
@@ -422,6 +422,20 @@ bool parse_fragment(RWMol &mol, ptree &frag,
       } else if (bond.display == "WedgedHashBegin" ||
                  bond.display == "WedgedHashEnd") {
         bnd->setBondDir(Bond::BondDir::BEGINWEDGE);
+      } else if (bond.display == "Wavy") {
+        switch(bond.getBondType()) {
+            case Bond::BondType::SINGLE:
+              bnd->setBondDir(Bond::BondDir::UNKNOWN);
```
Comment (greglandrum) [inline Code/GraphMol/FileParsers/CDXMLParser.cpp:428]: This is probably not part of this PR, but the mol file parser sets the `_MolFileBondCfg` property on bonds which have wedging information. This is then used by the CXSMILES writer and in `reapplyMolBlockWedging()`. It's worth considering doing that for the CDXML parser as well at some future point.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**